### DISCUSSION TITLE
The subject mapping is not working correctly in BMGS

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#211] [General Discussion]
**Example:**
I've 2 boards:
Board 1 : having 1 medium, 1 grade and 2 subjects.
```
board1
medium1
grade1
subject1
subject2
```
and Board 2 : having 1 medium, 1 grade and 2 subjects.
```
board2
medium2
grade2
subject3
subject4
```
When I'm selecting board2, getting following data in the drop downs:
**Medium** -> medium2
**Class** -> grade2
**Subject** -> subject1, subject2, subject3, subject4

But **subject** should display only subject3, subject4

**Association:**
I've done association using bottom up approach i.e, Subject -> Grade -> Medium -> Board

cc: @swayangjit @santoshgandham @sharathkashyap @luckynvdu @baljeetsingh @kameshbhr

**Comment by swayangjit** [comment]
@nidhikachhawa24 Please send the complete respone or curl of framework response

**Reply by nidhikachhawa24** [reply]
@swayangjit 
Here is the CURL of Framework:

**RESPONSE**
`{
    "id": "api.framework.read",
    "ver": "1.0",
    "ts": "2022-11-07T11:06:08.992Z",
    "params": {
        "resmsgid": "2eadae00-5e8c-11ed-a1a9-6174b1706874",
        "msgid": "2eac0050-5e8c-11ed-ae06-517a261a81f7",
        "status": "successful",
        "err": null,
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "framework": {
            "identifier": "devsandboxfw",
            "code": "devsandboxfw",
            "name": "Dev Framework",
            "description": "Dev Sandbox Framework Demo",
            "categories": [
                {
                    "identifier": "devsandboxfw_board",
                    "code": "board",
                    "terms": [
                        {
                            "identifier": "devsandboxfw_board_devsandboxfw",
                            "code": "devsandboxfw",
                            "translations": null,
                            "name": "Alpha Board",
                            "description": "Alpha Board",
                            "index": 1,
                            "category": "board",
                            "status": "Live"
                        }
                    ],
                    "translations": null,
                    "name": "Board",
                    "description": "board",
                    "index": 1,
                    "status": "Live"
                },
                {
                    "identifier": "devsandboxfw_medium",
                    "code": "medium",
                    "terms": [
                        {
                            "associations": [
                                {
                                    "identifier": "devsandboxfw_board_devsandboxfw",
                                    "code": "devsandboxfw",
                                    "translations": null,
                                    "name": "Alpha Board",
                                    "description": "Alpha Board",
                                    "index": 0,
                                    "category": "board",
                                    "status": "Live"
                                }
                            ],
                            "identifier": "devsandboxfw_medium_hindi",
                            "code": "hindi",
                            "translations": null,
                            "name": "Hindi",
                            "description": "Hindi",
                            "index": 1,
                            "category": "medium",
                            "status": "Live"
                        },
                        {
                            "associations": [
                                {
                                    "identifier": "devsandboxfw_board_devsandboxfw",
                                    "code": "devsandboxfw",
                                    "translations": null,
                                    "name": "Alpha Board",
                                    "description": "Alpha Board",
                                    "index": 0,
                                    "category": "board",
                                    "status": "Live"
                                }
                            ],
                            "identifier": "devsandboxfw_medium_english",
                            "code": "english",
                            "translations": null,
                            "name": "English",
                            "description": "English",
                            "index": 2,
                            "category": "medium",
                            "status": "Live"
                        }
                    ],
                    "translations": null,
                    "name": "Medium",
                    "description": "medium",
                    "index": 2,
                    "status": "Live"
                },
                {
                    "identifier": "devsandboxfw_gradelevel",
                    "code": "gradeLevel",
                    "terms": [
                        {
                            "associations": [
                                {
                                    "identifier": "devsandboxfw_medium_english",
                                    "code": "english",
                                    "translations": null,
                                    "name": "English",
                                    "description": "English",
                                    "index": 0,
                                    "category": "medium",
                                    "status": "Live"
                                },
                                {
                                    "identifier": "devsandboxfw_board_devsandboxfw",
                                    "code": "devsandboxfw",
                                    "translations": null,
                                    "name": "Alpha Board",
                                    "description": "Alpha Board",
                                    "index": 0,
                                    "category": "board",
                                    "status": "Live"
                                },
                                {
                                    "identifier": "devsandboxfw_medium_hindi",
                                    "code": "hindi",
                                    "translations": null,
                                    "name": "Hindi",
                                    "description": "Hindi",
                                    "index": 0,
                                    "category": "medium",
                                    "status": "Live"
                                }
                            ],
                            "identifier": "devsandboxfw_gradelevel_grade12",
                            "code": "grade12",
                            "translations": null,
                            "name": "Grade 12",
                            "description": "Grade 12",
                            "index": 1,
                            "category": "gradeLevel",
                            "status": "Live"
                        }
                    ],
                    "translations": null,
                    "name": "Grade",
                    "description": "grade",
                    "index": 3,
                    "status": "Live"
                },
                {
                    "identifier": "devsandboxfw_subject",
                    "code": "subject",
                    "terms": [
                        {
                            "associations": [
                                {
                                    "identifier": "devsandboxfw_gradelevel_grade12",
                                    "code": "grade12",
                                    "translations": null,
                                    "name": "Grade 12",
                                    "description": "Grade 12",
                                    "index": 0,
                                    "category": "gradeLevel",
                                    "status": "Live"
                                },
                                {
                                    "identifier": "devsandboxfw_medium_hindi",
                                    "code": "hindi",
                                    "translations": null,
                                    "name": "Hindi",
                                    "description": "Hindi",
                                    "index": 0,
                                    "category": "medium",
                                    "status": "Live"
                                },
                                {
                                    "identifier": "devsandboxfw_board_devsandboxfw",
                                    "code": "devsandboxfw",
                                    "translations": null,
                                    "name": "Alpha Board",
                                    "description": "Alpha Board",
                                    "index": 0,
                                    "category": "board",
                                    "status": "Live"
                                }
                            ],
                            "identifier": "devsandboxfw_subject_hindi",
                            "code": "hindi",
                            "translations": null,
                            "name": "Hindi",
                            "description": "Hindi",
                            "index": 1,
                            "category": "subject",
                            "status": "Live"
                        },
                        {
                            "associations": [
                                {
                                    "identifier": "devsandboxfw_gradelevel_grade12",
                                    "code": "grade12",
                                    "translations": null,
                                    "name": "Grade 12",
                                    "description": "Grade 12",
                                    "index": 0,
                                    "category": "gradeLevel",
                                    "status": "Live"
                                },
                                {
                                    "identifier": "devsandboxfw_medium_english",
                                    "code": "english",
                                    "translations": null,
                                    "name": "English",
                                    "description": "English",
                                    "index": 0,
                                    "category": "medium",
                                    "status": "Live"
                                },
                                {
                                    "identifier": "devsandboxfw_board_devsandboxfw",
                                    "code": "devsandboxfw",
                                    "translations": null,
                                    "name": "Alpha Board",
                                    "description": "Alpha Board",
                                    "index": 0,
                                    "category": "board",
                                    "status": "Live"
                                }
                            ],
                            "identifier": "devsandboxfw_subject_english",
                            "code": "english",
                            "translations": null,
                            "name": "English",
                            "description": "English",
                            "index": 2,
                            "category": "subject",
                            "status": "Live"
                        }
                    ],
                    "translations": null,
                    "name": "Subject",
                    "description": "subject",
                    "index": 4,
                    "status": "Live"
                }
            ],
            "type": "K-12",
            "objectType": "Framework"
        }
    }
}`

**Reply by swayangjit** [reply]
@nidhikachhawa24 I can see there are 2 subjects Hindi and English. What all subjects you are getting ?

**Reply by nidhikachhawa24** [reply]
@swayangjit I'm able to see all the subjects in subject dropdown.

Here is what I've selected:
**BOARD:** Alpha Board
**MEDIUM:** Hindi
**GRADE:** Class 12

But in Subject it should only show hindi, but I'm able to see Hindi and English both.

**Reply by nidhikachhawa24** [reply]
@swayangjit 
Sharing table of how the data should be displayed:

Board | Medium | Grade | Subject
-- | -- | -- | --
Alpha Board | Hindi | Grade 12 | Hindi
Alpha Board | English | Grade 12 | English

If I've selected **Hindi medium**, then the grade dropdown will show **Grade 12** and subject dropdown should show only **Hindi**.
But subject dropdown is showing **Hindi and English**.
